src/olympia/git/tests/test_tasks.py

```diff
@@ -162,3 +163,21 @@ def test_calls_extract_version_to_git_n_times(
                 mock.call(version_id=3, force_extraction=True),
             ]
         )
+
+
+class TestDeleteSourceGitRepositories(TestCase):
+    def test_deletes_source_repositories(self):
+        deleted_addon = addon_factory()
```

Comment:
I assume we're doing this to test the `with_deleted` kwarg? If so, I'm not sure how just using `1, 2, 3` as addon_ids is working without creating those addons first as well. Are they assumed to exist already, or does it not need there to be actual addons for those ids? I'm just a bit confused about how this test works. Also, does `addon_factory()` yield a deleted addon, and if not, why are we calling this `deleted_addon`?

Reply:
mm, no you're right. This is a mistake.

The task does not really care about the add-ons being created in the database because the add-on IDs are enough to create/delete the git repositories on disk. I think `deleted_addon` should be deleted here.